### ISSUE TITLE
High CPU usage in Idle | Version: 21.0

### ISSUE THREAD

**Issue by L0g4n** [nzbget/nzbget#680]
Hi,

NZBGet shows the behaviour of posing relatively high CPU usage in Idle (when it's supposed to do nothing) and thus high energy usage (bad for mobile devices) in Version 21.0:
![nzbget-21 0](https://user-images.githubusercontent.com/7620524/74935963-86aa1400-53e9-11ea-9555-260e3eced0a9.png)
This is running on Windows 10 1909 AMD64 on a Intel Core i7 4790k @ 4.3 GHz and it reports up to 16% CPU usage here.

According to "message system" there is nothing going on that justifies this high CPU usage.

**Edit**: NzbDirInterval is set to 5 seconds.

**Comment by biltongza** [comment]
I am seeing the same issue on FreeNAS.

```
last pid: 53152;  load averages:  3.80,  2.02,  0.99   up 20+18:12:34  11:33:24
5 processes:   1 running, 4 sleeping
CPU: 25.0% user,  0.1% nice,  1.2% system,  0.0% interrupt, 73.7% idle
Mem: 267M Active, 2932M Inact, 12G Wired, 506M Free
ARC: 6758M Total, 1293M MFU, 4660M MRU, 1132K Anon, 62M Header, 742M Other
     5047M Compressed, 5872M Uncompressed, 1.16:1 Ratio
Swap: 6144M Total, 6144M Free

  PID USERNAME    THR PRI NICE   SIZE    RES STATE   C   TIME    WCPU COMMAND
20975 nzbget       14  37    0 46968K 14420K uwait   3   2:26 100.13% nzbget
53152 root          1  20    0  7924K  3128K CPU2    2   0:00   0.01% top
21059 root          1  20    0  7488K  3872K pause   1   0:00   0.00% csh
20982 root          1  20    0  6460K  2332K nanslp  1   0:00   0.00% cron
20920 root          1  20    0  6420K  2456K select  3   0:00   0.00% syslogd
```

Downgrading to 20.0 resolved this for me.

**Comment by L0g4n** [comment]
Yeah, the energy usage is so high that on Windows 10 the sleep mode is prevented when nzbget is idling.
I once tried the current testing version where this phenomenon does not occur anymore.

**Comment by rteegarden** [comment]
Seeing the same issue, as soon as I go back to v20 then no issues.  On Windows with 4 core CPU, I see 25% usage (basically 100%) from nzbget.  I don't see anything strange in the logs.

![image](https://user-images.githubusercontent.com/1249082/82156542-67782280-9830-11ea-95d6-bd9d4bc1f811.png)

**Comment by markheloking** [comment]
Seems like this was fixed in https://github.com/nzbget/nzbget/commit/81aa56324f0418c15ee611e2508c39cfd6ee156b if I'm seeing this right. If so this looks like it can be closed.

**Comment by DarkAnakin** [comment]
Hi,
Not sure if it is correlate but I notice that if I add more than one news-server, nzbget cpu usage skyrocket to 100% even if the server are set to non active.
As example
```
Server2.Active=no
Server2.Name=xxxxxx
Server2.Level=0
Server2.Optional=no
Server2.Group=0
Server2.Host=xxxxx
Server2.Port=563
Server2.Username=xxxxxx
Server2.Password=xxxxxxx
Server2.JoinGroup=no
Server2.Encryption=yes
Server2.Cipher=
Server2.Connections=20
Server2.Retention=0
Server2.IpVersion=auto
Server2.Notes=
```

**Comment by Mange** [comment]
I have the same issue. I've even attached `strace` to all the processes related to nzbget and there isn't even much activity there, just the loop with sleep intervals every second.

I've disabled all servers, have emptied all logs, queues, partial files, history, and I have nothing in my queue. No installed extra scripts, and no schedule.

…and still, immediately after starting it goes to 100% CPU on one of the userland threads - one that doesn't even have any `strace` activity at all. Nothing shows up in any logs, either on stdout/stderr or in the log file.

I've disabled all scanning on incoming dirs, and have verified that all used directories are only used by `nzgbet`.

It's running with `-s` instead of `-D` because that's the only option as packaged using LinuxServer's docker images.

I've had to shut the process down for now.
